ion-hash/tests/ion_hash_tests.rs

```diff
@@ -257,3 +289,76 @@ fn seq_to_bytes<E: Element>(elem: &E) -> Vec<u8> {
         })
         .collect()
 }
+
+type IonHashTestResult<T> = Result<T, IonHashTestError>;
+
+#[derive(Debug, Error)]
+pub enum IonHashTestError {
+    /// Indicates that an IO error was encountered while reading or writing.
+    #[error("{source:?}")]
+    IonRsError {
+        #[from]
+        source: ion_rs::IonError,
+    },
+
+    #[error("FAIL: {test_case_name}: {message:?}")]
+    TestFailed {
+        test_case_name: String,
+        message: Option<String>,
+    },
+
+    #[error("ERROR: {test_case_name:?}: {message:?} {cause:?}")]
+    TestError {
+        test_case_name: Option<String>,
+        message: Option<String>,
+        cause: Option<Box<dyn Debug>>,
+    },
+
+    #[error("FAIL: {causes:?}")]
+    MultipleTestsFailed { causes: Vec<IonHashTestError> },
+}
+
+mod unknown_symbol_text_tests {
+    use super::*;
+    use ion_rs::Symbol;
+
+    #[test]
+    fn test_unknown_symbol_text() {
+        let unknown_symbol = Element::from(Symbol::unknown_text());
+        let digest = IdentityDigest::hash_element(&unknown_symbol).unwrap();
+
+        let expected_string = format!("{:02x?}", &[0x0b, 0x71, 0x0e]);
+        let actual_string = format!("{:02x?}", without_trailing_zeros(&digest[..]));
+        assert_eq!(expected_string, actual_string)
+    }
+
+    #[test]
+    fn test_unknown_field_name_text() {
+        let fields_itr = vec![(Symbol::unknown_text(), Element::from(1))];
+        let struct_ = owned::Struct::from_iter(fields_itr);
+        let element = Element::from(struct_);
+
+        let digest = IdentityDigest::hash_element(&element).unwrap();
+
+        let expected_string = format!(
+            "{:02x?}",
+            &[0x0b, 0xd0, 0x0c, 0x0b, 0x71, 0x0c, 0x0e, 0x0c, 0x0b, 0x20, 0x01, 0x0c, 0x0e, 0x0e]
```

Comment:
I wrote this out to understand it and double-check:
```rust
// s(struct) -> B || TQ || escape(concat(sort(H(field1), H(field2), ..., H(fieldn)))) || E
// s(field)  -> h(s(fieldname) || s(fieldvalue))
// s(scalar) -> B || TQ || escape(representation) || E

&[
  0x0b, 0xd0, // B(struct) || TQ(struct) || escape(
  0x0c, 0x0b, //   B(fieldname1/symbol)
  0x71,       //     TQ(symbol) empty symbol representation
  0x0c, 0x0e, //   E(fieldname1/symbol)
  0x0c, 0x0b, //   B(fieldvalue1/int)
  0x20, 0x01, //     TQ(int) || representation(1)
  0x0c, 0x0e, //   E(fieldvalue1/int) )
  0x0e        // E(struct)
]
```

Reply:
Yeah, I think that's it. I just copied these expected hashes directly from the corresponding test case in `ion-hash-test`, so it didn't occur to me to add an explanatory note like that.

Reply:
I wonder if it's possible/helpful to make an expression (macros?) that would look something like:
```rust
block(tq(STRUCT), escape(block(tq(EMPTY_SYMBOL)), block(tq(INTEGER), int_repr(1))))
```